lib/spack/spack/directory_layout.py

```diff
@@ -260,9 +260,10 @@ def check_installed(self, spec):
             return None
 
         if not os.path.isfile(spec_file_path):
-            raise InconsistentInstallDirectoryError(
-                'Install prefix exists but contains no spec.yaml:',
-                "  " + path)
+            return None
+#            raise InconsistentInstallDirectoryError(
+#                'Install prefix exists but contains no spec.yaml:',
+#                "  " + path)
```

Comment:
#2172 was merged 14 days ago, I'm working with develop from 1 day ago.  So... #2172 has not fixed the problem this change to directory_layout.py is addressing.

This change addresses #2319.  I put it in the same PR that addresses #2316 because both bugs must be addressed in order for `spack setup` to work.  I agree, there might be better ways to fix #2319.  But I don't know of what that way would be.  Absent a better solution ,we should merge this PR, at least temporarily, if we want `spack setup` working today.  I don't want someone at the tutorial trying `spack setup` and having it crash on them very first thing.

If we can find a better way to fix #2319 in the future, I'd be happy to revert this PR and do the better way instead.  Let's not let perfect become the enemy of good enough.